components/turret.py

```diff
@@ -107,110 +81,70 @@ def scan(self, heading):
-            self.target_count = 0
-
-    ################
-    #
-    # The remainder of this file, as well as the FINDING_INDEX state and all
-    # the relevant references, can be removed once we no longer explicitly
-    # do indexing.
-    #
-    ################
-
-    # Seek for 200ms at first, before reversing and doubling
-    STARTING_MAX_TICKS = 10
-
-    # The indexing does not use the encoder at all right now. It just counts
-    # the times it's called, incrementing (or decrementing) a tick count each
-    # time. This method resets the counting mechanism.
-    def _reset_ticks(self) -> None:
-        self.motor_speed = 0.2  # Flips when seeking, so not a constant
-        self.seeking = False
-        self.tick_count = 0
-        self.max_ticks = self.STARTING_MAX_TICKS
-        self.max_ticks_factor = -2
-        self.tick_increment = 1
-
-    # Find the nearest index and reset the encoder
-    def run_indexing(self) -> None:
-        self._reset_ticks()
-        self.current_state = self.FINDING_INDEX
-
-    def _do_indexing(self):
-        # Are we there yet? If so, greb the encoder value, stop the motor,
-        # and stop seeking
-        if self._index_found() == Index.CENTRE:
-            self.baseline_azimuth = self.motor.getSelectedSensorPosition(0)
-            self.baseline_is_provisional = False
-            self.motor.stopMotor()
-            self._reset_ticks()
-            self.current_state = self.IDLE
-            # self.logger.info("Found the sensor")
-            return
-
-        # If we haven't started yet, start seeking
-        if not self.seeking:
-            self.motor.set(ctre.ControlMode.PercentOutput, self.motor_speed)
-            self.tick_count = 0
-            self.max_ticks = self.STARTING_MAX_TICKS
-            self.seeking = True
-            # self.logger.info("starting to seek")
-            return
-
-        self.tick_count = self.tick_count + self.tick_increment
-        if self.tick_count == self.max_ticks:
-            self.motor.stopMotor()
-            self.motor_speed = -self.motor_speed
-            self.motor.set(ctre.ControlMode.PercentOutput, self.motor_speed)
-            self.tick_increment = self.tick_increment * -1
-            self.max_ticks = self.max_ticks * self.max_ticks_factor
-            return
-
-        # Currently running and haven't hit the limit nor found an index.
-        # Just keep the motor running
-        self.motor.set(ctre.ControlMode.PercentOutput, self.motor_speed)
+            _reset_encoder(0)
+        # TODO: Repeat for other index marks
+
+    def _reset_encoder(self, counts) -> None:
+        current_count = self.motor.getSelectedSensorPosition()
+        current_target = self.motor.getClosedLoopTarget()
+        delta = current_target - current_count
+        self.motor.setSelectedSensorPosition(counts)
+        # Reset any current target using the new absolute azimuth
+        self._slew_to_counts(counts + delta)
+        self.has_index = True
```

Comment:
has_index is defined above as a function, but is here being set to a boolean. I suspect you mean self.index_found. But that then begs the question of why we want a function at all. This one isn't called at all, anywhere right now. I don't think whether the turret has found an index should be part of it's API, as whether or not it knows exactly where it is is an internal implementation detail about how scanning works, and not something any controller should care about.

Reply:
Correct.
Fixed.